Idea to use embedded resources on Windows and AppImage builds.
Run qmake with RESOURCES+='data.qrc' DEFINES+='APPIMAGE_BUILD=1'

--- a/sources/helpdialog.cpp
+++ b/sources/helpdialog.cpp
@@ -32,6 +32,13 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "settings.h"
 
+#if defined(Q_OS_WIN) || defined(APPIMAGE_BUILD)
+
+#undef DATA_INSTALL_PATH
+#define DATA_INSTALL_PATH QDir(":/data")
+
+#endif // Q_OS_WIN || APPIMAGE_BUILD
+
 namespace qpdfview
 {
 
--- a/sources/main.cpp
+++ b/sources/main.cpp
@@ -65,6 +65,13 @@ const char* __attribute__((used)) stack_cookie = "\0$STACK:500000\0";
 
 #endif // __amigaos4__
 
+#if defined(Q_OS_WIN) || defined(APPIMAGE_BUILD)
+
+#undef DATA_INSTALL_PATH
+#define DATA_INSTALL_PATH QDir(":/data")
+
+#endif // Q_OS_WIN || APPIMAGE_BUILD
+
 namespace
 {
 
--- /dev/null
+++ b/data.qrc
@@ -0,0 +1,69 @@
+<RCC>
+  <qresource prefix="/data">
+    <file>data/help.html</file>
+    <file>data/help_az.html</file>
+    <file>data/help_bg.html</file>
+    <file>data/help_cs.html</file>
+    <file>data/help_de.html</file>
+    <file>data/help_en_GB.html</file>
+    <file>data/help_es.html</file>
+    <file>data/help_fa.html</file>
+    <file>data/help_fi.html</file>
+    <file>data/help_fr.html</file>
+    <file>data/help_gl.html</file>
+    <file>data/help_it.html</file>
+    <file>data/help_ja.html</file>
+    <file>data/help_ko.html</file>
+    <file>data/help_lt.html</file>
+    <file>data/help_ms.html</file>
+    <file>data/help_pl.html</file>
+    <file>data/help_pt_BR.html</file>
+    <file>data/help_pt.html</file>
+    <file>data/help_ru.html</file>
+    <file>data/help_sk.html</file>
+    <file>data/help_sv.html</file>
+    <file>data/help_uk.html</file>
+    <file>data/help_vi.html</file>
+    <file>data/help_zh_CN.html</file>
+    <file>data/qpdfview_ast.qm</file>
+    <file>data/qpdfview_az.qm</file>
+    <file>data/qpdfview_be.qm</file>
+    <file>data/qpdfview_bg.qm</file>
+    <file>data/qpdfview_bs.qm</file>
+    <file>data/qpdfview_ca.qm</file>
+    <file>data/qpdfview_cs.qm</file>
+    <file>data/qpdfview_da.qm</file>
+    <file>data/qpdfview_de.qm</file>
+    <file>data/qpdfview_el.qm</file>
+    <file>data/qpdfview_en_GB.qm</file>
+    <file>data/qpdfview_eo.qm</file>
+    <file>data/qpdfview_es.qm</file>
+    <file>data/qpdfview_eu.qm</file>
+    <file>data/qpdfview_fi.qm</file>
+    <file>data/qpdfview_fr.qm</file>
+    <file>data/qpdfview_gl.qm</file>
+    <file>data/qpdfview_he.qm</file>
+    <file>data/qpdfview_hr.qm</file>
+    <file>data/qpdfview_hu.qm</file>
+    <file>data/qpdfview_id.qm</file>
+    <file>data/qpdfview_it.qm</file>
+    <file>data/qpdfview_kk.qm</file>
+    <file>data/qpdfview_ky.qm</file>
+    <file>data/qpdfview_lt.qm</file>
+    <file>data/qpdfview_ms.qm</file>
+    <file>data/qpdfview_my.qm</file>
+    <file>data/qpdfview_pl.qm</file>
+    <file>data/qpdfview_pt_BR.qm</file>
+    <file>data/qpdfview_pt.qm</file>
+    <file>data/qpdfview_ro.qm</file>
+    <file>data/qpdfview_ru.qm</file>
+    <file>data/qpdfview_sk.qm</file>
+    <file>data/qpdfview_sv.qm</file>
+    <file>data/qpdfview_th.qm</file>
+    <file>data/qpdfview_tr.qm</file>
+    <file>data/qpdfview_ug.qm</file>
+    <file>data/qpdfview_uk.qm</file>
+    <file>data/qpdfview_vi.qm</file>
+    <file>data/qpdfview_zh_CN.qm</file>
+  </qresource>
+</RCC>
